src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -373,7 +373,7 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
                     Action<Diagnostic> addExceptionDiagnostic = diagnostic => analyzerExceptionDiagnostics.Add(diagnostic);
                     var analyzerOptions = new AnalyzerOptions(ImmutableArray<AdditionalText>.CastUp(additionalTextFiles));
                     analyzerDriver = AnalyzerDriver.CreateAndAttachToCompilation(compilation, analyzers, analyzerOptions, analyzerManager, addExceptionDiagnostic, Arguments.ReportAnalyzer, out compilation, analyzerCts.Token);
-                    getAnalyzerDiagnostics = () => analyzerDriver.GetDiagnosticsAsync().Result;
+                    getAnalyzerDiagnostics = () => analyzerDriver.GetDiagnosticsAsync(compilation).Result;
```

Comment:
Actually, this API was intended to be async only for public consumers of diagnostics queue from the analyzer driver (when we planned to make it public). Right now its only consumer is the command line compiler, which invokes it after all compilation events have been generated, so we can probably convert the API to sync API. I will do so in a later refactoring.